test/remote/gateways/remote_redsys_rest_test.rb

```diff
@@ -0,0 +1,206 @@
+    assert_success capture
+    assert_match(/Refund.*approved/, capture.message)
+  end
+
+  def test_successful_authorize_and_failed_capture
+    authorize = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success authorize
+    assert_equal 'Transaction Approved', authorize.message
+    assert_not_nil authorize.authorization
+
+    capture = @gateway.capture(2 * @amount, authorize.authorization, @options)
+    assert_failure capture
+    assert_match(/SIS0062 ERROR/, capture.message)
+  end
+
+  def test_failed_authorize
+    response = @gateway.authorize(@amount, @declined_card, @options)
+    assert_failure response
+    assert_equal 'Refusal with no specific reason', response.message
+  end
+
+  def test_successful_void
+    authorize = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success authorize
+
+    void = @gateway.void(authorize.authorization, @options)
+    assert_success void
+    assert_equal '100', void.params['ds_amount']
+    assert_equal 'Cancellation Accepted', void.message
+  end
+
+  def test_failed_void
+    authorize = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success authorize
+
+    authorization = "#{authorize.params[:ds_order]}|#{@amount}|203"
+    void = @gateway.void(authorization, @options)
+    assert_failure void
+    assert_equal 'SIS0027 ERROR', void.message
+  end
+
+  def test_successful_verify
+    assert response = @gateway.verify(@credit_card, @options)
+    assert_success response
+
+    assert_equal 'Transaction Approved', response.message
+    assert_success response.responses.last, 'The void should succeed'
+    assert_equal 'Cancellation Accepted', response.responses.last.message
+  end
+
+  def test_unsuccessful_verify
+    assert response = @gateway.verify(@declined_card, @options)
+    assert_failure response
+    assert_equal 'Refusal with no specific reason', response.message
+  end
+
+  def test_transcript_scrubbing
+    transcript = capture_transcript(@gateway) do
+      @gateway.purchase(@amount, @credit_card, @options)
+    end
+    clean_transcript = @gateway.scrub(transcript)
+
+    assert_scrubbed(@gateway.options[:secret_key], clean_transcript)
+    assert_scrubbed(@credit_card.number, clean_transcript)
+    assert_scrubbed(@credit_card.verification_value.to_s, clean_transcript)
+  end
+
+  def test_transcript_scrubbing_on_failed_transactions
+    transcript = capture_transcript(@gateway) do
+      @gateway.purchase(@amount, @declined_card, @options)
+    end
+    clean_transcript = @gateway.scrub(transcript)
+
+    assert_scrubbed(@gateway.options[:secret_key], clean_transcript)
+    assert_scrubbed(@credit_card.number, clean_transcript)
+    assert_scrubbed(@credit_card.verification_value.to_s, clean_transcript)
+  end
+
+  def test_encrypt_handles_url_safe_character_in_secret_key_without_error
+    gateway = RedsysRestGateway.new({
+      login: '091952713',
```

Comment:
Is there a reason we are using different hard coded creds here instead of the creds from the fixtures?

Reply:
Yep! This test is making sure url unsafe characters are fine to be used in the `login`/`authentication`